packages/conferer/src/Conferer/FromConfig/Internal.hs

```diff
@@ -63,12 +63,11 @@ class FromConfig a where
   -- partial overriding.
   fetchFromConfig :: Key -> Config -> IO a
   default fetchFromConfig :: (Typeable a, Generic a, IntoDefaultsG (Rep a), FromConfigG (Rep a)) => Key -> Config -> IO a
-  fetchFromConfig k c = do
-    let config =
-          case fetchFromDefaults @a k c of
-            Just d -> c & addDefaults (intoDefaultsG k $ from d)
-            Nothing -> c
-    to <$> fetchFromConfigG k config
+  fetchFromConfig = allowingFetchOverride $ \key config -> do
+    let configWithDefaults = case fetchFromDefaults @a key config of
+          Just d -> config & addDefaults (intoDefaultsG key $ from d)
+          Nothing -> config
+    to <$> fetchFromConfigG key configWithDefaults
```

Comment:
Do we want to always allow fetch override? Because if that's the case, why not have:
```
fetchFromConfig = allowingFetchOverride fetchFromConfig'
fetchFromConfig' = ...
```

And have people override `fetchFromConfig'`?

I think that breaks the interface tho :'(. But are there other places were `allowingFetchOverride` could be added? Because as it is now it needs to be added in a lot of places (and in a lot of future places when we add new FetchFromConfigs too D: )

Reply:
this comment is outdated